beesdoo_portal_shift/controllers/main.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf8 -*-
+from datetime import datetime
+from math import floor
+
+from openerp import http
+from openerp.http import request
+
+class Sandbox(http.Controller):
+
+    @http.route('/shift_irregular_worker', auth='public', website=True)
+    def shift_irregular_worker(self, **kwargs):
+        # Get all the shifts in the future with no worker
+        now = datetime.now()
+        shifts = request.env['beesdoo.shift.shift'].search(
+            [('start_time', '>', now.strftime("%Y-%m-%d %H:%M:%S")),
+            ('worker_id', '=', False)],
+            order="start_time, task_template_id, task_type_id",
+        )
+        # Loop on all the shifts
+        shift_templates = []
+        current_template = None
+        current_shift_template = None
+        current_remaining_space = 0
+        for shift in shifts:
+            # For a planning id, count the number of shift that don't
+            # have a worker.
+            if shift.task_template_id == current_template:
+                # If we are in the same template then update the number
+                # of available sapce
```

Comment:
typo on available "space"

Reply:
put place instead

Reply:
Ok, I see my typo error. But I think "palce" is for a "location". English speaker use "space" or "room" to say "spot occupied by someone".